### PR TITLE
network: show knwon route protocol name nicely in debugging logs

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/network/networkd-route.c [hunk 1]
```diff
@@ -846,6 +846,8 @@ static const char * const route_protocol_table[] = {
         [RTPROT_KERNEL] = "kernel",
         [RTPROT_BOOT]   = "boot",
         [RTPROT_STATIC] = "static",
+        [RTPROT_RA]     = "ra",
+        [RTPROT_DHCP]   = "dhcp",
```
Comment (ssahani) [inline src/network/networkd-route.c]: hmm these two should not configurable. Should be set by the code itself .

File: man/systemd.network.xml [hunk 2]
```diff
@@ -1225,8 +1225,8 @@
           <term><varname>Protocol=</varname></term>
           <listitem>
             <para>The protocol identifier for the route. Takes a number between 0 and 255 or the special values
-            <literal>kernel</literal>, <literal>boot</literal> and <literal>static</literal>. Defaults to
-            <literal>static</literal>.
+            <literal>kernel</literal>, <literal>boot</literal>, <literal>static</literal>,
+            <literal>ra</literal> and <literal>dhcp</literal>. Defaults to <literal>static</literal>.
```
Comment (ssahani) [inline man/systemd.network.xml:1229]: We should not add it here in man. The debugging messages should automatically pick it . So adding in the table just fine. Not in the man.

Comment (yuwata) [reply man/systemd.network.xml:1229]: Initially I thought so. But considering about `kernel` and `boot`, I neither have idea about the situation we need to set these values. Moreover, we can still set numeric values which are equivalent to `ra` or `dhcp`. So, I think there are no reason to list `ra` and `dhcp` in the man page.

Comment (ssahani) [reply man/systemd.network.xml:1229]: Which value we can set ? This is not how this works.

Comment (yuwata) [reply man/systemd.network.xml:1229]: Arbitrary value by numeric.

Comment (ssahani) [reply man/systemd.network.xml:1229]: This is actually letting the admins doing wrong configs. These two no meant for configurable . I do not agree with this patch.

**Comment by yuwata** [comment]
OK. Now, debug message shows known protocol name, but config parser still accepts only `kernel`, `boot` and `static`. @ssahani PTAL.

**Comment by ssahani** [comment]
Yeah that make sense.  something like ```static``` are  configurable and some are not because those are supposed to configurable by the protocol who sets this.

**Comment by yuwata** [comment]
Thanks. Setting the green label.

**Pull request merged**